src/systems/filecoin_blockchain/storage_power_consensus/_index.md

```diff
@@ -44,70 +46,168 @@ More precisely,
+
+In turn these random seeds are used by:
+
+- The {{<sref sector_sealer>}} as SealSeeds to bind sector commitments to a given subchain.
+- The {{<sref post_generator>}} as PoStChallenges to prove sectors remain committed as of a given block.
+- The Storage Power subsystem as randomness in {{<sref leader_election>}} to determine their eligibility to mine a block.
+
+This randomness may be drawn from various Filecoin chain epochs by the respective protocols that use them according to their security requirements.
+
+It is important to note that a given Filecoin network and a given drand network need not have the same round time, i.e. blocks may be generated faster or slower by Filecoin than randomness is generated by drand.
+For instance, if the drand beacon is producing randomness twice as fast as Filecoin produces blocks, we might expect two random values to be produced in a Filecoin epoch, conversely if the Filecoin network is twice as fast as drand, we might expect a random value every other Filecoin epoch. Accordingly, depending on both networks' configurations, certain Filecoin blocks could contain multiple or no drand entries.
+
+In all cases, Filecoin blocks must include all drand beacon outputs generated since the last epoch in the `BeaconEntries` field of the block header. Any use of randomness from a given Filecoin epoch should use the last valid drand entry included in a Filecoin block, as follows:
+
+```go
+// Note that the below function is a temporary solution and likely to be
+// updated in future. 
+GetBeaconEntryForEpoch(epoch) BeaconEntry {
+    var rand Randomness
+    for rand == nil {
+        seeds := GetBeaconEntriesForEpoch(epoch)
+        if seeds == nil {
+            epoch -= 1
+        } else {
+            rand = seeds[len(seeds)-1].Randomness
+        }
+    }
+    return rand
+}
+
+GetBeaconEntriesForEpoch(epoch) []BeaconEntry {
+
+    // special case genesis: the genesis block is pre-generated and so cannot include a beacon entry 
+    // (since it will not have been generated). Hence, we only start checking beacon entries at the first block after genesis.
+    // If that block includes a wrong beacon entry, we simply assume that a majority of honest miners at network birth will
+    // simply fork.
+    entries := []
+    if epoch == 0 {
+        return entries
+    }
+
+    maxDrandRound := MaxBeaconRoundForEpoch(epoch)
+
+    // if checking the first post-genesis block, simply fetch the latest entry.
+    if epoch == 1 {
+        rand := drand.Public(maxDrandRound)
+        return append(entries, rand)
+    }
+
+    // for the rest, fetch all drand entries generated between this epoch and last
+    prevMaxDrandRound := MaxBeaconRoundForEpoch(epoch - 1)
+    if (maxDrandRound == prevMaxDrandRound) {
+        // no new beacon randomness
+        return entries
+    }
+
+    entries := []
+    curr := maxDrandRound
+    for curr > prevMaxDrandRound {
+        rand := drand.Public(curr)
```

Comment:
We must make the assumption that this call is blocking: i.e. if the drand network is DOWN, then this call BLOCKS.

Reply:
added as a comment. thank you!